src/main/java/servlets/EditReceiptServlet.java

```diff
@@ -0,0 +1,125 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.Gson;
+import com.google.sps.servlets.UploadReceiptServlet.InvalidDateException;
+import com.google.sps.servlets.UploadReceiptServlet.InvalidPriceException;
+import java.io.IOException;
+import java.time.Clock;
+import java.util.Arrays;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet with a POST handler that updates an existing receipt entity in Datastore.
+ */
+@WebServlet("/edit-receipt")
+public class EditReceiptServlet extends HttpServlet {
+  private static final String USER_NOT_LOGGED_IN_WARNING =
+      "User must be logged in to edit a receipt.";
+
+  // Logs to System.err by default.
+  private static final Logger logger = Logger.getLogger(UploadReceiptServlet.class.getName());
+
+  private final DatastoreService datastore;
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final Clock clock;
+
+  public EditReceiptServlet() {
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.clock = Clock.systemDefaultZone();
+  }
+
+  public EditReceiptServlet(DatastoreService datastore, Clock clock) {
+    this.datastore = datastore;
+    this.clock = clock;
+  }
+
+  /**
+   * Updates the receipt entity with the given ID using the properties sent in the request body from
+   * the receipt analysis page.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (!userService.isUserLoggedIn()) {
+      logger.warning(USER_NOT_LOGGED_IN_WARNING);
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      response.getWriter().println(USER_NOT_LOGGED_IN_WARNING);
+      return;
+    }
+
+    Entity receipt;
+    try {
+      long id = Long.parseLong(request.getParameter("id"));
```

Comment:
If you want the receipt analysis page to send an ID, you'll have to modify upload.js to send the ID when it redirects to the receipt analysis page.

Reply:
Added ID in #88.